sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -184,6 +190,21 @@ class __SYCL_EXPORT SYCLMemObjT : public SYCLMemObjI {
     };
   }
 
+  template <typename Destination>
+  __SYCL_DLL_LOCAL EnableIfOutputIteratorBool<Destination>
+  set_final_data(Destination FinalData) {
+    MUploadDataFunctor = [this, FinalData]() {
+      using DestinationValueT = iterator_value_type_t<Destination>;
+      // TODO if Destination is ContiguousIterator then don't create
+      // ContiguousStorage. updateHostMemory works only with pointer to
+      // continuous data.
+      const size_t Size = MSizeInBytes / sizeof(DestinationValueT);
+      bool *ContiguousStorage = new bool[Size];
+      updateHostMemory(ContiguousStorage);
+      std::copy(ContiguousStorage, ContiguousStorage + Size, FinalData);
```

Comment:
I'm not sure it'll work correctly as layout of `vector<bool>` may vary with implementation of standard library.
Hence, testing is needed.

Reply:
I tested it and it works